### PR TITLE
[PORT HIL-SERL] Better unit tests coverage for SAC policy

### PR DESCRIPTION
**Pull request by helper2424**
## What this does
Explain what this PR does. Feel free to tag your PR with the appropriate label(s).

Examples:
|  Title               | Label           |
|----------------------|-----------------|
| Fixes #[issue]       | (🐛 Bug)        |
| Adds new dataset     | (🗃️ Dataset)    |
| Optimizes something  | (⚡️ Performance) |

## How it was tested
Explain/show how you tested your changes.

Examples:
- Added `test_something` in `tests/test_stuff.py`.
- Added `new_feature` and checked that training converges with policy X on dataset/environment Y.
- Optimized `some_function`, it now runs X times faster than previously.

## How to checkout & try? (for the reviewer)
Provide a simple way for the reviewer to try out your changes.

Examples:
```bash
pytest -sx tests/test_stuff.py::test_something
```
```bash
python lerobot/scripts/train.py --some.option=true
```

## SECTION TO REMOVE BEFORE SUBMITTING YOUR PR
**Note**: Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR. Try to avoid tagging more than 3 people.

**Note**: Before submitting this PR, please read the [contributor guideline](https://github.com/huggingface/lerobot/blob/main/CONTRIBUTING.md#submitting-a-pull-request-pr).

### CODE REVIEW

**Comment by helper2424** [comment]
Its WIP

**Comment by helper2424** [comment]
Ok, finally finished tests for modeling part.

File: lerobot/common/policies/sac/configuration_sac.py [hunk 1]
```diff
@@ -140,7 +140,6 @@ class SACConfig(PreTrainedConfig):
     )
 
     # Architecture specifics
-    camera_number: int = 1
```
Comment (helper2424) [inline lerobot/common/policies/sac/configuration_sac.py:143]: We don't use this parameter

File: lerobot/common/policies/sac/configuration_sac.py [hunk 2]
```diff
@@ -184,6 +183,9 @@ class SACConfig(PreTrainedConfig):
     actor_learner_config: ActorLearnerConfig = field(default_factory=ActorLearnerConfig)
     concurrency: ConcurrencyConfig = field(default_factory=ConcurrencyConfig)
 
+    # Optimizations
+    use_torch_compile: bool = True
```
Comment (helper2424) [inline lerobot/common/policies/sac/configuration_sac.py:187]: Let's allow to skip using torch compiler.

File: lerobot/common/policies/sac/modeling_sac.py [hunk 3]
```diff
@@ -473,9 +476,11 @@ def _init_actor(self, continuous_action_dim):
             encoder_is_shared=self.shared_encoder,
             **asdict(self.config.policy_kwargs),
         )
-        if self.config.target_entropy is None:
+
+        self.target_entropy = self.config.target_entropy
+        if self.target_entropy is None:
             dim = continuous_action_dim + (1 if self.config.num_discrete_actions is not None else 0)
-            self.config.target_entropy = -np.prod(dim) / 2
```
Comment (helper2424) [inline lerobot/common/policies/sac/modeling_sac.py:478]: Let's avoid modifying the configuration.  It seems that ti should be immutable and all params changes should be done inside the modeling object. 

If one config will be reused with several modeling instances - we will see many strange bugs.

File: lerobot/common/policies/sac/modeling_sac.py [hunk 4]
```diff
@@ -997,14 +1002,6 @@ def orthogonal_init():
     return lambda x: torch.nn.init.orthogonal_(x, gain=1.0)
 
 
-class Identity(nn.Module):
```
Comment (helper2424) [inline lerobot/common/policies/sac/modeling_sac.py:1000]: We don't use - so I have dropped the class.

Comment (AdilZouitine) [reply lerobot/common/policies/sac/modeling_sac.py:1000]: perfect

File: lerobot/common/policies/sac/configuration_sac.py (hunk 1 above)
Comment (AdilZouitine) [reply lerobot/common/policies/sac/configuration_sac.py:143]: you're right we didn't use it anymore

**Review by AdilZouitine — COMMENTED** [review]
lgtm
